### PR TITLE
 Log macros and verbosity level change for commit messages

### PR DESCRIPTION
**Pull request by arichiardi**
As per #557.

### CODE REVIEW

File: boot/pod/src/boot/util.clj [hunk 1]
```diff
@@ -57,6 +57,14 @@
       (print ((or color identity) (apply format args)))
       (flush))))
 
+(defmacro print**
+  "Macro version of boot.util/print* but arguments are only evaluated
+  when the message will be printed."
+  [verbosity fmt & args]
+  `(when (>= @*verbosity* ~verbosity)
+     (binding [*out* *err*]
+       (print (ansi/bold-cyan (format ~fmt ~@args))))))
```
Comment (arichiardi) [inline boot/pod/src/boot/util.clj]: Uhm, maybe the color part can be improved here (haven't noticed that immediately, maybe cause it's late :smile:)

**Comment by arichiardi** [comment]
My rigorous testing :smile: 

2017-01-05-224223_114x199_scrot

```
  (boot.util/trace "Fn %s\n" 1)
  (boot.util/dbug "Fn %s\n" 1)
  (boot.util/info "Fn %s\n" 1)
  (boot.util/warn "Fn %s\n" 1)
  (boot.util/fail "Fn %s\n" 1)
  (boot.util/trace* "Macro %s\n" 1)
  (boot.util/dbug* "Macro %s\n" 1)
  (boot.util/info* "Macro %s\n" 1)
  (boot.util/warn* "Macro %s\n" 1)
  (boot.util/fail* "Macro %s\n" 1)
```

**Comment by micha** [comment]
Looks good, thanks @arichiardi! Can we make the trace level output cyan (maybe not bold)? That would help differentiate it from stdout.

**Comment by arichiardi** [comment]
We sure can 😀

**Comment by arichiardi** [comment]
@micha @alandipert Done ;)

**Comment by arichiardi** [comment]
Another idea, should we make `Aether` messages `trace` now?

File: boot/aether/src/boot/aether.clj [hunk 2]
```diff
@@ -68,7 +68,7 @@
 
 (defn transfer-listener
   [info]
-  (util/dbug "Aether: %s\n" (with-out-str (pprint/pprint info)))
+  (util/trace* "Aether: %s\n" (with-out-str (pprint/pprint info)))
```
Comment (rfkm) [inline boot/aether/src/boot/aether.clj]: FYI: 
This should fix the bad network throughput. (ref. #565)
In my environment, it takes about 10 minutes to fetch 20MB jar file with `dbug`, while it takes only 20 seconds with `dbug*`. I tried to make a PR by myself but I will wait for this PR to be merged.
BTW, it seems that this code was introduced by [this conflict resolution commit](https://github.com/boot-clj/boot/commit/de16fb099dac2795a5e314038e5edf2d07b92807#diff-b547159e61e59485c59eccc8b7d5f7a1).

Comment (Deraen) [inline boot/aether/src/boot/aether.clj]: Yep, macro version is definitively a good call here, else the pprint is called for the info even if dbug prints are disabled.

Comment (arichiardi) [inline boot/aether/src/boot/aether.clj]: @rfkm can you by any chance clone and confirm that this solves?

Comment (rfkm) [inline boot/aether/src/boot/aether.clj]: @arichiardi I've just tested with this branch and confirmed the `trace*` version is up to 100 times faster than the `dbug` version 👍

**Pull request merged**